OpenSim/Common/Component.cpp

```diff
@@ -780,53 +782,8 @@ Array<std::string> Component::getStateVariableNames() const
             stateNames.append(pathName + "/" + subStateNames[i]);
         }
     }
-*/
-
-    // Include the states of its subcomponents
-    for (unsigned int i = 0; i<_memberSubcomponents.size(); i++) {
-        Array<std::string> subnames = _memberSubcomponents[i]->getStateVariableNames();
-        int nsubs = subnames.getSize();
-        const std::string& subCompName = _memberSubcomponents[i]->getName();
-        std::string::size_type front = subCompName.find_first_not_of(" \t\r\n");
-        std::string::size_type back = subCompName.find_last_not_of(" \t\r\n");
-        std::string prefix = "";
-        if (back >= front) // have non-whitespace name
-            prefix = subCompName + "/";
-        for (int j = 0; j<nsubs; ++j) {
-            names.append(prefix + subnames[j]);
-        }
-    }
-    for(unsigned int i=0; i<_propertySubcomponents.size(); i++){
-        Array<std::string> subnames = _propertySubcomponents[i]->getStateVariableNames();
-        int nsubs = subnames.getSize();
-        const std::string& subCompName =  _propertySubcomponents[i]->getName();
-        // TODO: We should implement checks that names do not have whitespace at the time 
-        // they are assigned and not here where it is a waste of time - aseth
-        std::string::size_type front = subCompName.find_first_not_of(" \t\r\n");
-        std::string::size_type back = subCompName.find_last_not_of(" \t\r\n");
-        std::string prefix = "";
-        if(back >= front) // have non-whitespace name
-            prefix = subCompName+"/";
-        for(int j =0; j<nsubs; ++j){
-            names.append(prefix+subnames[j]);
-        }
-    }
-
-    for (unsigned int i = 0; i<_adoptedSubcomponents.size(); i++) {
-        Array<std::string> subnames = _adoptedSubcomponents[i]->getStateVariableNames();
-        int nsubs = subnames.getSize();
-        const std::string& subCompName = _adoptedSubcomponents[i]->getName();
-        std::string::size_type front = subCompName.find_first_not_of(" \t\r\n");
-        std::string::size_type back = subCompName.find_last_not_of(" \t\r\n");
-        std::string prefix = "";
-        if (back >= front) // have non-whitespace name
-            prefix = subCompName + "/";
-        for (int j = 0; j<nsubs; ++j) {
-            names.append(prefix + subnames[j]);
-        }
-    }
 
```

Comment:
Can you explain how the new block (that used to be commented out) accomplishes the function of the removed one? It sure looks cleaner.

Reply:
The iterator traverses the three separately maintained private list of subcomponents. Teh comment was added before we had the iterator working efficiently and should have been updated long ago. I noticed it while making this change and decided to finish it.